cfme/tests/ansible/test_embedded_ansible_services.py

```diff
@@ -39,6 +42,69 @@
 ]
 
 
+@pytest.fixture()
+def ansible_linked_vm_action(
+    appliance, ansible_catalog_item, new_vm
+):
+    with update(ansible_catalog_item):
+        ansible_catalog_item.provisioning = {"playbook": "add_single_vm_to_service.yml"}
+
+    action_collection = appliance.collections.actions
+
+    action_values = {
+        "run_ansible_playbook": {
+            "playbook_catalog_item": ansible_catalog_item.name,
+            "inventory": {"specific_hosts": True, "hosts": new_vm.ip_address},
+        }
+    }
+
+    action = action_collection.create(
+        fauxfactory.gen_alphanumeric(),
+        action_type="Run Ansible Playbook",
+        action_values=action_values,
+    )
+    yield action
+
+    action.delete_if_exists()
+
+
+@pytest.fixture()
+def new_vm(provider, big_template):
+    vm_collection = provider.appliance.provider_based_collection(provider)
+    vm = vm_collection.instantiate(random_vm_name(context='ansible'),
+                                  provider,
+                                  template_name=big_template.name)
+    vm.create_on_provider(find_in_cfme=True)
+    logger.debug(
+        "Fixture new_vm set up! Name: %r Provider: %r", vm.name, vm.provider.name
+    )
+    yield vm
+    vm.cleanup_on_provider()
+
+
+@pytest.fixture()
+def ansible_policy_linked_vm(appliance, new_vm, ansible_linked_vm_action):
+    policy = appliance.collections.policies.create(
+        VMControlPolicy,
+        fauxfactory.gen_alpha(),
+        scope="fill_field(VM and Instance : Name, INCLUDES, {})".format(new_vm.name),
+    )
+    policy.assign_actions_to_event(
+        "Tag Complete", [ansible_linked_vm_action.description]
+    )
+    policy_profile = appliance.collections.policy_profiles.create(
+        fauxfactory.gen_alpha(), policies=[policy]
+    )
+    new_vm.assign_policy_profiles(policy_profile.description)
+    yield
+
+    if policy.exists:
+        policy.unassign_events("Tag Complete")
+        new_vm.unassign_policy_profiles(policy_profile.description)
+        policy_profile.delete()
+        policy.delete()
```

Comment:
optional:
can we try `delete_if_exists()` for both

Reply:
@digitronik as we are removing assigned event couldn't able to use `delete_if_exists` directly there.

Reply:
Oops my bad I didn't see that included `BaseEntity` there, thanks @digitronik for pointing out.

Reply:
I still think this applies, we should check that the policy exists before deleting

Reply:
Already updated with `policy_profile.delete_if_exists()`

Reply:
@sbulage what about the policy though?

Reply:
@jawatts yes for Policy `delete_if_exists()` is available but here we are checking if policy is there then first unassign event and later deleting it.

Reply:
@john-dupuy @mshriver adding `policy.delete_if_exists()` is breaking tests at teardown. Adding back above lines to have test case working.

Reply:
Never Mind. Solved locator issue after debugging.